extensions/amp-redbull-player/0.1/amp-redbull-player.js

```diff
@@ -0,0 +1,298 @@
+      this.tagId_
+    }&locale=${this.locale_}`;
+    const src = `${origin}${params}`;
+
+    const name = JSON.stringify(this.getMetadata_());
+    this.iframe_ = disableScrollingOnIframe(
+      createFrameFor(this, src, name, SANDBOX)
+    );
+
+    this.unlistenFrame_ = listen(this.win, 'message', this.boundOnMessage_);
+    const loaded = this.loadPromise(this.iframe_).then(() => {
+      this.onReady_();
+    });
+    return loaded;
+  }
+
+  /** @private */
+  onReady_() {
+    const {element} = this;
+    Services.videoManagerForDoc(element).register(this);
+
+    this.sendCommand_({msg: 'amp-loaded'});
+    element.dispatchCustomEvent(VideoEvents.LOAD);
+  }
+
+  /**
+   * @return {!JsonObject}
+   * @private
+   */
+  getMetadata_() {
+    const {sourceUrl, canonicalUrl} = Services.documentInfoForDoc(this.element);
+
+    return dict({
+      'sourceUrl': sourceUrl,
+      'canonicalUrl': canonicalUrl,
+    });
+  }
+
+  /**
+   * Sends a command to the player through postMessage.
+   * @param {string} command
+   * @private
+   */
+  sendCommand_(command) {
+    if (this.iframe_ && this.iframe_.contentWindow) {
+      if (!command.id) {
+        command.id = `${TAG}-${this.tagId_}`;
+      }
+      this.postMessage_(command);
+    }
+  }
+
+  /** @override */
+  unlayoutCallback() {
+    this.removeIframe_();
+    return true; // layout again.
+  }
+
+  /** @private */
+  removeIframe_() {
+    if (this.iframe_) {
+      removeElement(this.iframe_);
+      this.iframe_ = null;
+    }
+    if (this.unlistenFrame_) {
+      this.unlistenFrame_();
+      this.unlistenFrame_ = null;
+    }
+  }
+
+  /**
+   * @param {!Event} event
+   * @private
+   */
+  originMatches_(event) {
+    return originMatches(event, this.iframe_, /.*/);
+  }
+
+  /**
+   * @param {!Event} event
+   * @private
+   */
+  onMessage_(event) {
+    if (!this.iframe_) {
+      return;
+    }
+
+    if (!this.originMatches_(event)) {
+      return;
+    }
+
+    const eventData = getData(event);
+    if (!isJsonOrObj(eventData)) {
+      return;
+    }
+
+    const data = objOrParseJson(eventData);
+
+    if (data.id === `redbull-amp-video-tracking-${this.tagId_}`) {
+      const type = ANALYTICS_EVENT_TYPE_PREFIX + data.type;
+      this.dispatchCustomAnalyticsEvent_(type, data);
+    }
+  }
+
+  /**
+   * @param {string} eventType
+   * @param {!Object<string, string>=} vars
+   */
+  dispatchCustomAnalyticsEvent_(eventType, vars = {}) {
+    userAssert(
+      getAnalyticsEventTypePrefixRegex().test(eventType),
+      'Invalid analytics `eventType`. Value must start with `%s`.',
+      ANALYTICS_EVENT_TYPE_PREFIX
+    );
+    this.element.dispatchCustomEvent(
+      VideoEvents.CUSTOM_TICK,
+      dict({
+        'eventType': `video-custom-tracking-${this.tagId_}`,
+        'vars': vars,
+      })
+    );
+  }
+
+  /**
+   * @param {!JsonObject} message
+   * @private
+   */
+  postMessage_(message) {
+    if (!message.msg) {
```

Comment:
Use a string property accessor `message['msg']` instead. This is required since we minify property names on build time, but the shape of `JsonObject`s is not minifyable. 

By using bracket-string syntax we tell the compiler that the property name should not be obfuscated.

I'm also a bit confused as to why this check is needed?

Reply:
removed this check